src/adapter/src/coord/timestamp_oracle.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! An interface/trait that provides write and read timestamps, reads observe
+//! exactly their preceding writes.
+//!
+//! Specifically, all read timestamps will be greater or equal to all previously reported completed
+//! write timestamps, and strictly less than all subsequently emitted write timestamps.
+
+use async_trait::async_trait;
+
+use crate::coord::timeline::WriteTimestamp;
+
+pub mod consensus;
+pub mod durable;
```

Comment:
maybe a better name for this than "durable" is "catalog"? they're all durable, no?